packages/go_router/lib/src/router.dart

```diff
@@ -302,12 +305,64 @@ class GoRouter extends ChangeNotifier implements RouterConfig<RouteMatchList> {
   /// Navigate to a URI location w/ optional query parameters, e.g.
   /// `/family/f2/person/p1?color=blue`
   void go(String location, {Object? extra}) {
-    log.info('going to $location');
-    routeInformationProvider.go(location, extra: extra);
+    if (routerDelegate.isOnExitBeingCalled) {
```

Comment:
it is weird the goRouter is mutating routerDelegate's state.

Here are list of possible navigation I can think of
1. GoRouter.go/push/pop
2. GoRouter.backButtonDispatcher (called when android bachbutton, which in terms calls RouterDelegate.popRoute)
3. RouterDelegate._handlePopPageWithRouteMatch(can be called from (2), or people call Navigator.pop directly)
4. GoRouteInformationProvider.didPushRouteInformation and didPushRoute (moible/web deeplinks, browser backward and forward button).

If we are going to block them all, I think we need to create a lock object to be owned by `GoRouter` and passed to GoRouteInformationProvider and GoRouterDelegate (or RootBackButtonDispatcher if we want to block the back button at this level).

Reply:
also we may want to throw if people doing things that trigger any of above 4 item during their onExit.